spatialdata/_core/_spatialdata.py

```diff
@@ -242,12 +280,391 @@ def _init_add_element(self, name: str, element_type: str, overwrite: bool) -> za
+                # error 0 (we refer to this in the tests)
+                raise RuntimeError("No path found between the two coordinate systems")
+            elif len(paths) > 1:
+                if intermediate_coordinate_systems is None:
+                    # if one and only one of the paths has lenght 1, we choose it straight away, otherwise we raise
+                    # an expection and ask the user to be more specific
+                    paths_with_length_1 = [p for p in paths if len(p) == 2]
+                    if len(paths_with_length_1) == 1:
+                        path = paths_with_length_1[0]
+                    else:
+                        # error 1
+                        raise RuntimeError(
+                            "Multiple paths found between the two coordinate systems. Please specify an intermediate "
+                            "coordinate system."
+                        )
+                else:
+                    if has_type_spatial_element(intermediate_coordinate_systems):
+                        intermediate_coordinate_systems = id(intermediate_coordinate_systems)
+                    paths = [p for p in paths if intermediate_coordinate_systems in p]
+                    if len(paths) == 0:
+                        # error 2
+                        raise RuntimeError(
+                            "No path found between the two coordinate systems passing through the intermediate"
+                        )
+                    elif len(paths) > 1:
+                        # error 3
+                        raise RuntimeError(
+                            "Multiple paths found between the two coordinate systems passing through the intermediate"
+                        )
+                    else:
+                        path = paths[0]
+            else:
+                path = paths[0]
+            transformations = []
+            for i in range(len(path) - 1):
+                transformations.append(g[path[i]][path[i + 1]]["transformation"])
+            sequence = Sequence(transformations)
+            return sequence
+
+    def filter_by_coordinate_system(self, coordinate_system: str) -> SpatialData:
+        """
+        Filter the SpatialData by a coordinate system.
+
+        Parameters
+        ----------
+        coordinate_system
+            The coordinate system to filter by.
+
+        Returns
+        -------
+        The filtered SpatialData.
+        """
+        elements: dict[str, dict[str, SpatialElement]] = {}
+        for element_type, element_name, element in self._gen_elements():
+            transformations = self.get_all_transformations(element)
+            if coordinate_system in transformations:
+                if element_type not in elements:
+                    elements[element_type] = {}
+                elements[element_type][element_name] = element
+        return SpatialData(**elements, table=self.table)
+
+    def transform_element_to_coordinate_system(
```

Comment:
I would remove this as API wise I don't super get it, will elaborate in comment

Reply:
Ok thanks, answering in the comment.

Reply:
I have removed these functions and added new global methods with better signature (I put them in `_spatialdata.py`, but probably we want to move them somewhere when we will refactor the import structure of the subpackages.